### PR TITLE
5078 r30 add group

### PR DESCRIPTION
**Pull request by AnumehaSrivastava05**

### CODE REVIEW

File: frontend/src/main.js [hunk 1]
```diff
@@ -56,6 +55,8 @@ function initApp() {
   library.add(faSpinner)
   library.add(faTimes)
   library.add(faQuestionCircle)
+  library.add(faPlus)
```
Comment (weskubo-cgi) [inline frontend/src/main.js:58]: I don't think we are using faPlus. To improve performance and minimize size we don't want to include anything that's unused.

File: frontend/src/main.js [hunk 2]
```diff
@@ -1,22 +1,21 @@
 import '@bcgov/bc-sans/css/BCSans.css'
 
-import { createApp } from 'vue'
-
-import { library } from '@fortawesome/fontawesome-svg-core'
-import { faCheckCircle, faChevronDown, faExclamationCircle, faExclamationTriangle, faInfoCircle, faSpinner, faTimes, faQuestionCircle } from '@fortawesome/free-solid-svg-icons'
-import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+import { faCheckCircle, faChevronDown, faExclamationCircle, faExclamationTriangle, faInfoCircle, faPlus, faQuestionCircle, faSpinner, faTimes, faTrashAlt } from '@fortawesome/free-solid-svg-icons'
```
Comment (weskubo-cgi) [inline frontend/src/main.js:3]: I don't think we are using faPlus. To improve performance and minimize size we don't want to include anything that's unused.

File: frontend/src/util/validators.js [hunk 3]
```diff
@@ -25,7 +24,7 @@ export function validatePHN(phn) {
   return validatePHNFormat(phn)
 }
 
-function validatePHNFormat(phn) {
+export function validatePHNFormat(phn) {
```
Comment (weskubo-cgi) [inline frontend/src/util/validators.js:27]: I don't think this needs to be exported. It was meant as a "private" function and we aren't using in AddGroupMember anymore.

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 4]
```diff
@@ -0,0 +1,313 @@
+<template>
+  <div id="addGroupMember" v-if="addMode">
+    <form @submit.prevent="submitForm">
+      <AppRow>
+        <AppCol class="col3">
+          <AppInput :e-model="v$.groupNumber" id="groupNumber" label="Group Number" type="text" v-model.trim="groupNumber"/>
+        </AppCol>
+          <AppCol class="col3">
+          <AppDateInput :e-model="v$.coverageEffectiveDate" id="dp-input-coverageEffectiveDate" label="Coverage Effective Date" tooltip tooltipText="Date always defaults to first day of the month"
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/AddGroupMember.vue:9]: The id shouldn't have "dp-input-" as a prefix. This is something that the datepicker component will assign.

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+        }
+        
+      } catch (err) {
+        this.$store.commit('alert/setErrorAlert', `${err}`)
+      } finally {
+        this.submitting = false
+      }
+    },
+    addDependent(dependentPHN) {    
+      this.dependents.push(dependentPHN);
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/AddGroupMember.vue:245]: Trailing semicolon not required.

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+          <AppInput :e-model="v$.groupMemberNumber" id="groupMemberNumber" label="Group Member Number (Optional)" type="text" v-model.trim="groupMemberNumber" />
+        </AppCol>
+      </AppRow>
+      <AppRow>
+        <AppCol class="col3">
+          <AppInput :e-model="v$.departmentNumber" id="departmentNumber" label="Department Number (Optional)" type="text" v-model="departmentNumber" />
+        </AppCol>
+      </AppRow>
+      <AppRow>
+        <AppCol class="col4">
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/AddGroupMember.vue:29]: Can we make this col3 for consistency with other fields?

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+        </AppCol>
+      </AppRow>
+       <AppRow>
+        <AppCol class="col7">
+          <AppInput :e-model="v$.homeAddress.addressLine4" id="addressLine4" label="Line 4 (Optional)" type="text" v-model="homeAddress.addressLine4" />
+        </AppCol>
+      </AppRow>
+      <AppRow>
+        <AppCol class="col4">
+          <AppInput :e-model="v$.homeAddress.postalCode" id="postalCode" label="Postal Code" type="text" v-model.trim="homeAddress.postalCode" />
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/AddGroupMember.vue:55]: Can we make this col3 for consistency with other fields?

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+          <AppInput :e-model="v$.mailingAddress.addressLine3" id="mailingAddress3" label="Line 3 (Optional)" v-model="mailingAddress.addressLine3" />
+        </AppCol>
+      </AppRow>
+      <AppRow>
+        <AppCol class="col7">
+          <AppInput :e-model="v$.mailingAddress.addressLine4" id="mailingAddress4" label="Line 4 (Optional)" v-model="mailingAddress.addressLine4" />
+        </AppCol>
+      </AppRow>
+      <AppRow>
+        <AppCol class="col4">
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/AddGroupMember.vue:79]: Can we make this col3 for consistency with other fields?

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+          <AppInput :e-model="v$.departmentNumber" id="departmentNumber" label="Department Number (Optional)" type="text" v-model="departmentNumber" />
+        </AppCol>
+      </AppRow>
+      <AppRow>
+        <AppCol class="col4">
+          <AppInput :e-model="v$.telephone" id="telephone" label="Telephone (Optional)" type="text" v-model.trim="telephone" placeholder="1234567890" />
+        </AppCol>
+      </AppRow>
+      <AppRow>
+        <AppCol class="col7">
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/AddGroupMember.vue:34]: I think all the address fields can be col6.

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+          </AppCol> 
+        </AppRow>      
+      </div> 
+      <AddListDependent :dependents="dependents" @add-dependent="addDependent" @remove-dependent="removeDependent"/>
+      <AppRow>
+        <AppButton :submitting="submitting" mode="primary" type="submit">Submit</AppButton>
+        <AppButton @click="resetForm()" mode="secondary" type="button">Clear</AppButton>
+      </AppRow>
+    </form>
+  <br />
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/AddGroupMember.vue:114]: Indentation is off.

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+    },
+  },
+  methods: {
+    async submitForm() {
+      this.submitting = true,
+      this.addOk = false, 
+      this.addMode= true
+      try {
+        const isValid = await this.v$.$validate()
+        console.log(isValid)
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/AddGroupMember.vue:187]: Please remove console.log once development is complete.

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+      this.mailingAddress.addressLine3 = ''
+      this.mailingAddress.addressLine4 = ''
+      this.mailingAddress.postalCode = ''
+      this.spousePhn = ''
+      this.dependents = []
+      this.v$.$reset()
+      this.$store.commit('alert/dismissAlert')
+      this.submitting = false,
+      this.addOk = false, 
+      this.addMode = true
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/AddGroupMember.vue:273]: this.result should be reset as well.

File: frontend/tests/e2e/pages/groupmember/AddGroupMember.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
+import { Selector } from 'testcafe';
```
Comment (weskubo-cgi) [inline frontend/tests/e2e/pages/groupmember/AddGroupMember.js:1]: Trailing commas aren't required in this file.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/views/groupmember/AddGroupMember.vue [hunk 14]
```diff
@@ -0,0 +1,313 @@
...
+        </AppRow>
+        <AppRow>
+          <AppCol class="col4">
+          </AppCol> 
+        </AppRow>      
+      </div> 
+      <AddListDependent :dependents="dependents" @add-dependent="addDependent" @remove-dependent="removeDependent"/>
+      <AppRow>
+        <AppButton :submitting="submitting" mode="primary" type="submit">Submit</AppButton>
+        <AppButton @click="resetForm()" mode="secondary" type="button">Clear</AppButton>
```
Comment (daveb-hni) [inline frontend/src/views/groupmember/AddGroupMember.vue:111]: Clear button doesn't clear content in the Dependent PHN field but not sure if this is significant as this data is not submitted directly.

**Review by weskubo-cgi — APPROVED** [review]
Changes look good.

**Pull request merged**